### ISSUE TITLE
Error with CloudQueue + CloudFlow

### ISSUE THREAD

**Issue by dsyme** [mbraceproject/MBrace.Core#136]
This gives a serialization error, at least with Thespian:
```
let requestQueue = CloudQueue.New<int>() |> cluster.Run
let outputQueue = CloudQueue.New<int>() |> cluster.Run

let processingFlow = 
    CloudFlow.OfCloudQueue(requestQueue, 4)
    |> CloudFlow.toCloudQueue outputQueue
    |> cluster.CreateProcess

processingFlow.ShowInfo() 

processingFlow.Result
```

giving
```
Nessos.FsPickler.FsPicklerException: Error deserializing object of type 'MBrace.Runtime.Components.SiftedClosure`1[MBrace.Runtime.CloudProcessResult]'. ---> System.InvalidCastException: Object must implement IConvertible.
   at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider)
   at System.Runtime.Serialization.SerializationInfo.GetValue(String name, Type type)
   at Nessos.Thespian.ThespianException..ctor(SerializationInfo info, StreamingContext context)
   at ISerializableCtor(SerializationInfo , StreamingContext )
   at <StartupCode$FsPickler>.$ISerializablePickler.reader@201-37.Invoke(ReadState r, String _arg4)
   at Nessos.FsPickler.CompositePickler`1.Read(ReadState state, String tag)
   at Nessos.FsPickler.Pickler`1.UntypedRead(ReadState state, String tag)
   at Nessos.FsPickler.CompositePickler`1.Read(ReadState state, String tag)
   at Nessos.FsPickler.Pickler`1.UntypedRead(ReadState state, String tag)
   at <StartupCode$FsPickler>.$ISerializablePickler.reader@201-37.Invoke(ReadState r, String _arg4)
   at Nessos.FsPickler.CompositePickler`1.Read(ReadState state, String tag)
   at Nessos.FsPickler.Pickler`1.UntypedRead(ReadState state, String tag)
>    at Nessos.FsPickler.CompositePickler`1.Read(ReadState state, String tag)
   at classDeserializer(Pickler[] , ReadState )
   at Nessos.FsPickler.CompositePickler`1.Read(ReadState state, String tag)
   at unionDeserializer(Pickler[][] , UnionCaseSerializationHelper , ReadState )
   at Nessos.FsPickler.CompositePickler`1.Read(ReadState state, String tag)
   at unionDeserializer(Pickler[][] , UnionCaseSerializationHelper , ReadState )
   at Nessos.FsPickler.CompositePickler`1.Read(ReadState state, String tag)
   at Nessos.FsPickler.RootSerialization.readRootObject[T](IPicklerResolver resolver, ReflectionCache reflectionCache, IPickleFormatReader formatter, FSharpOption`1 streamingContext, FSharpOption`1 sifted, Pickler`1 pickler)
   --- End of inner exception stack trace ---
   at Nessos.FsPickler.RootSerialization.readRootObject[T](IPicklerResolver resolver, ReflectionCache reflectionCache, IPickleFormatReader formatter, FSharpOption`1 streamingContext, FSharpOption`1 sifted, Pickler`1 pickler)
   at Nessos.FsPickler.FsPicklerSerializer.Deserialize[T](Stream stream, FSharpOption`1 pickler, FSharpOption`1 streamingContext, FSharpOption`1 encoding, FSharpOption`1 leaveOpen)
   at <StartupCode$FsPickler>.$Serializer.f@169-4[T](FsPicklerSerializer bp, FSharpOption`1 pickler, FSharpOption`1 streamingContext, FSharpOption`1 encoding, Stream m)
   at Nessos.FsPickler.FsPicklerSerializer.UnPickle[T](Byte[] data, FSharpOption`1 pickler, FSharpOption`1 streamingContext, FSharpOption`1 encoding)
   at <StartupCode$MBrace-Runtime>.$PersistedValueManager.ReadPickleOrFileAsync@94.Invoke(Unit unitVar) in C:\Users\eirik\Development\mbrace\MBrace.Core\src\MBrace.Runtime\Components\PersistedValueManager.fs:line 96
   at Microsoft.FSharp.Control.AsyncBuilderImpl.callA@803.Invoke(AsyncParams`1 args)
   at MBrace.Core.Internals.ExceptionDispatchInfoUtils.Async.RunSync[T](FSharpAsync`1 workflow, FSharpOption`1 cancellationToken) in C:\Users\eirik\Development\mbrace\MBrace.Core\src\MBrace.Core\Utils\ExceptionDispatchInfo.fs:line 140
   at <StartupCode$FSI_0026>.$FSI_0026.main@() in C:\GitHub\mbraceproject\mbrace-docs\docs\starterkit\HandsOnTutorial\8-using-cloud-queues.fsx:line 70
```

**Comment by dsyme** [comment]
Actually this fails too 

```
let queue = CloudQueue.New<string>() |> cluster.Run

CloudQueue.Enqueue (queue, "hello") |> cluster.Run

let msg = CloudQueue.Dequeue(queue) |> cluster.Run
```

F# 3.1, versions below


```
    MBrace.Azure (0.12.2-beta)
      Argu (>= 1.0.0 < 2.0.0)
      FsPickler (>= 1.4.0)
      FsPickler.Json (>= 1.4.0)
      MBrace.Core (0.12.1-beta)
      MBrace.Runtime (0.12.1-beta)
      Newtonsoft.Json (>= 7.0.1)
      Vagabond (>= 0.9.1)
      WindowsAzure.ServiceBus (>= 2.5.2 < 2.6.0)
      WindowsAzure.Storage (>= 4.3.0 < 4.4.0)
    MBrace.Core (0.12.1-beta)
    MBrace.Flow (0.12.1-beta)
      MBrace.Core (0.12.1-beta)
      Streams (>= 0.3.0 < 0.4.0)
    MBrace.Runtime (0.12.1-beta)
      FsPickler (>= 1.4.0 < 1.5.0)
      FsPickler.Json (>= 1.4.0 < 1.5.0)
      MBrace.Core (0.12.1-beta)
      Vagabond (>= 0.9.1 < 0.10.0)
    MBrace.Thespian (0.12.1-beta)
      Argu (>= 1.0.0 < 2.0.0)
      FsPickler (>= 1.4.0 < 1.5.0)
      MBrace.Core (0.12.1-beta)
      MBrace.Runtime (0.12.1-beta)
      Thespian.Core (0.1.6-alpha)
      Vagabond (>= 0.9.1 < 0.10.0)
```

**Comment by eiriktsarpalis** [comment]
I can't reproduce with MBrace.Core 0.12.2. Is this possibly related the already fixed FSharp.Core issue that we discussed?

**Comment by dsyme** [comment]
Cool, it's fixed with 0.12.2
